spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraHealthIndicator.java

```diff
@@ -38,25 +39,24 @@
 	private static final SimpleStatement SELECT = SimpleStatement
 			.newInstance("SELECT release_version FROM system.local").setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
 
-	private CassandraOperations cassandraOperations;
-
-	public CassandraHealthIndicator() {
-		super("Cassandra health check failed");
-	}
+	private final CqlSession session;
 
 	/**
 	 * Create a new {@link CassandraHealthIndicator} instance.
-	 * @param cassandraOperations the Cassandra operations
+	 * @param session the {@link CqlSession}.
 	 */
-	public CassandraHealthIndicator(CassandraOperations cassandraOperations) {
+	public CassandraHealthIndicator(CqlSession session) {
 		super("Cassandra health check failed");
-		Assert.notNull(cassandraOperations, "CassandraOperations must not be null");
-		this.cassandraOperations = cassandraOperations;
+		Assert.notNull(session, "session must not be null");
+		this.session = session;
 	}
 
 	@Override
 	protected void doHealthCheck(Health.Builder builder) throws Exception {
-		String version = this.cassandraOperations.getCqlOperations().queryForObject(SELECT, String.class);
+		Row row = this.session.execute(SELECT).one();
+		Assert.notNull(row, "system.local should always return one row");
+		String version = row.getString(0);
+		Assert.notNull(version, "release_version should never be null");
```

Comment:
These assertions really will never fail as any C* node would report exactly one row with one non-null column for such a CQL query. But they make static analysis tools happy.

Reply:
There's no need to do that, you could check if the row is non null (since the API forces  you to do that check), something like:

```java
Row row = this.session.execute(SELECT).one();
builder.up();
if (row != null) {
	builder.withDetail("version", row.getString(0));
}
```

Reply:
Isn't that slightly different? In the very unlikely case where the row would be null, in my version the  health indicator would be set to DOWN, whereas in your case, it would be UP. But I will do as you suggest, since a null row is totally unrealistic anyways.

Reply:
Yes, you are right, I overlooked that. Although the "These assertions really will never fail" makes me think it won't make a difference.